### PR TITLE
SCC-4110: Modal replacement

### PR DESCRIPTION
**Pull request by 7emansell**
## Ticket:

- JIRA ticket [SCC-4110](https://jira.nypl.org/browse/SCC-4110)

## This PR does the following:

- Replaces all modals with "default" or "confirmation" modal from DS update

## How has this been tested?

Locally– check that all modals in My Account work as they did before, plus that there is the new "confirmation" type when you attempt to cancel a hold 

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by 7emansell** [comment]
Waiting on applied filter fix and update location modal work to be merged into main

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "coverage": "jest --coverage"
   },
   "dependencies": {
-    "@nypl/design-system-react-components": "3.1.1",
+    "@nypl/design-system-react-components": "3.1.1-modal-rc-3",
```
Comment (EdwinGuzman) [inline package.json:18]: @7emansell DS 3.1.3 was just release and it contains the latest `Modal` updates. Thanks for working on it!

File: src/components/MyAccount/RequestsTab/CancelButton.tsx [hunk 2]
```diff
@@ -10,6 +10,13 @@ import {
 } from "@nypl/design-system-react-components"
 import { BASE_URL } from "../../../config/constants"
 import styles from "../../../../styles/components/MyAccount.module.scss"
+import type {
+  BaseModalProps,
+  BaseProps,
+  ConfirmationModalProps,
+  DefaultModalProps,
+} from "@nypl/design-system-react-components/dist/src/components/Modal/Modal"
```
Comment (EdwinGuzman) [inline src/components/MyAccount/RequestsTab/CancelButton.tsx:18]: You should now be able to import these types directly from `"@nypl/design-system-react-components"`

Comment (EdwinGuzman) [reply src/components/MyAccount/RequestsTab/CancelButton.tsx:18]: `BaseProps` is not used in this file so you can remove that import.

**Comment by EdwinGuzman** [comment]
When you make this ready for review, make sure to update the changelog.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**